tests/acceptance/30_custom_promise_types/01_basic_module.cf

```diff
@@ -0,0 +1,31 @@
+promise agent example
+{
+    interpreter => "/bin/bash";
+    # FIXME variable expansion in promise block -> CFE-3460
+    path => "$(this.promise_dirname)/example_module.sh";
+}
+
+bundle agent main
+{
+  vars:
+    "test_string"
+      string => "hello, modules";
+    "test_file"
+      string => "$(this.promise_dirname)/test_file";
+
+  example:
+    cfengine::
+      "$(test_file)"
+        message => "$(test_string)";
+
+  classes:
+    "file_exists"
+      expression => fileexists("$(test_file)");
+    file_exists::
+      "content_is_correct"
+        expression => strcmp("$(test_string)", readfile("$(test_file)"));
+
+  reports:
+    file_exists.content_is_correct::
+      "$(this.promise_filename) Pass";
```

Comment:
@nickanderson might want to use some `dcs_()` stuff here to check the file and print the result.

Reply:
I try to avoid those when I can.

Reply:
the DCS stuff is good, that's how you get the soft fail support, and reasonable places for metadata.

Reply:
@nickanderson I don't like it because it obscures the logic. I like to be able to debug my tests without looking up what `dcs_whatever` actually does.